src/xma/src/xmaapi/xmaapi.cpp

```diff
@@ -211,7 +212,7 @@ void xma_thread1() {
         }
     }
     //Print all stats here
-    std::lock_guard<std::mutex> lock(g_xma_singleton->m_mutex);
+    std::shared_lock<std::shared_timed_mutex> lock(g_xma_singleton->m_mutex);
```

Comment:
```suggestion
    std::shared_lock lock(g_xma_singleton->m_mutex);
```

Reply:
template type deduction not available in c++14